### DISCUSSION TITLE
disable inlay hints

### DISCUSSION THREAD

**Discussion by andrey-zh** [LazyVim/LazyVim#3191] [Q&A] [answered]
hello, how i can disable inlay hints? 
i couldn't find in ui settings.
thanks
![image](https://github.com/LazyVim/LazyVim/assets/87668873/c7dc8f93-1a82-498e-8ee6-69240f17fe3c)

**Comment by folke** [comment]
That screenshot doesnt show folding?

**Reply by andrey-zh** [reply, marked as answer]
ment inlay hints :D 
found in the ui settings 
`leader + u + h`
or add in the config 
```lua
{
  "neovim/nvim-lspconfig",
  opts = {
    inlay_hints = { enabled = false },
  },
}
```
thanks!

**Reply by p4-k4** [reply]
Thanks!

**Reply by simonLeary42** [reply]
If I want to put this in my `~/.config/nvim/lua/config/options.lua`, I would want to `return` that lua table, right? I did this and sourced `options.lua` and it doesn't seem to work for me.

edit: this works for me, kind of:
```lua
vim.lsp.inlay_hint.enable(false)
```
if I `:source ~/.config/nvim/lua/config/options.lua`, the inlay hints go away, but they are enabled every time I open up a new nvim.

edit 2: I think the actual problem is that I should have returned that lua table from a file under `plugins/`, not in `config/options.lua`.

**Reply by aryandutt** [reply]
Yeah your edit 2 works.
return the above mentioned table from `~/.config/nvim/lua/plugins/lspconfig.lua` and it worked

**Reply by gitsang** [reply]
Add

```lua
{
  "neovim/nvim-lspconfig",
  opts = {
    inlay_hints = { enabled = false },
  },
}
``` 

to `~/.config/nvim/lua/plugins/lsp.lua` not work for me.

**Reply by OArnarsson** [reply]
> Add
> 
> ```lua
> {
>   "neovim/nvim-lspconfig",
>   opts = {
>     inlay_hints = { enabled = false },
>   },
> }
> ```
> 
> to `~/.config/nvim/lua/plugins/lsp.lua` not work for me.

You must return a lua table.
Prepend `return` to your code and it should work.

**Comment by pidanou** [comment]
Is there a way to disable them on lsp attach? It seems that they are toggled on by default which is a bit annoying. It seems that it is isnt opt-in but rather opt-out right now

**Reply by folke** [reply]
You can disable them with:

```lua
{
  "neovim/nvim-lspconfig",
  opts = {
    inlay_hints = { enabled = false },
  },
}
```

**Reply by pidanou** [reply]
I mean, i want them, but when i press `<leader>uh`, right not I always have them and i need to do `<leader>uh` to toggle off. Sorry if i was unclear

**Reply by dpetka2001** [reply]
If you do what Folke said in his previous comment, then they will be not enabled by default. And then you just press `<leader>uh` to toggle them on.

**Reply by pidanou** [reply]
Ok my bad I thought disabling it was entirely disabling the feature, thanks!

**Reply by alwindoss** [reply]
> You can disable them with:
> 
> ```lua
> {
>   "neovim/nvim-lspconfig",
>   opts = {
>     inlay_hints = { enabled = false },
>   },
> }
> ```

where do you add this?

**Reply by dpetka2001** [reply]
@alwindoss Create a file `~/.config/nvim/lua/plugins/whatever.lua` with contents
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      inlay_hints = { enabled = false },
    },
  },
}
```

**Reply by alwindoss** [reply]
This worked. Thanks!
